lib/logic/player/favorites.dart

```diff
@@ -65,16 +72,114 @@ class FavoritesControl with Control {
     _favoriteSetsMap.set(favoriteSet, key: contentType);
   }
 
+  /// Let the user resolve any conflicts between the current favorite state and the MediaStore state.
+  /// The [favoritesInMediaStore] are the source ids of the songs that are currently marked as favorites in the
+  /// MediaStore. The list will be updated if the user decides to mark more favorites or un-mark some favorites.
+  /// Returns `true` if the conflicts were resolved successfully, `false` if the user decided to cancel.
+  Future<bool> _resolveConflictsOnSwitchingToMediaStore(Set<int> favoritesInMediaStore) async {
+    final previousFavorites = _favoriteSetsMap.get(ContentType.song);
+    final newFavoritesFromMediaStore = favoritesInMediaStore
+        .difference(previousFavorites)
+        .map((id) =>
+            ContentControl.instance.getContent(ContentType.song).firstWhereOrNull((song) => song.sourceId == id))
+        .whereType<Song>() // Filter unknown songs
+        .toSet();
+    final newFavoritesFromLocalStore = previousFavorites
+        .difference(favoritesInMediaStore)
+        .map((id) =>
+            ContentControl.instance.getContent(ContentType.song).firstWhereOrNull((song) => song.sourceId == id))
+        .whereType<Song>() // Filter unknown songs
+        .toSet();
+    if (newFavoritesFromMediaStore.isNotEmpty || newFavoritesFromLocalStore.isNotEmpty) {
+      final context = AppRouter.instance.navigatorKey.currentContext;
+      if (context != null) {
+        final chosenFavorites =
+            await _showFavoriteConflictDialog(context, newFavoritesFromLocalStore, newFavoritesFromMediaStore);
+        if (chosenFavorites == null) {
+          await Settings.useMediaStoreForFavoriteSongs.set(false);
+          return false;
+        }
+        final songsToUnFavor = newFavoritesFromMediaStore.difference(chosenFavorites);
+        final songsToFavour = chosenFavorites.difference(newFavoritesFromMediaStore);
```

Comment:
let's be consistent with what dialect we use - we use American "favor" everwhere else 🙃
also, it should be "unfavor", no need to uppercase F

Here and everywhery else in this PR